httpcache_test.go

```diff
@@ -29,12 +24,11 @@ func (c *fakeClock) since(t time.Time) time.Duration {
 	return c.elapsed
 }
 
-var _ = Suite(&S{})
-
```

Comment:
Why just have a single TestAll? If you use separate exported TestXxx funcs (instead of testXxx), then you can use `-test.run=pattern` to selectively run certain tests, and you get more informative failure messages.

And you can just factor the common setup code in the top of the TestAll func to a helper func like `func setup() { ... }`.